src/server.rs

```diff
@@ -1,56 +1,34 @@
 use super::minecraft_protocol::read_var_int;
-use super::packet;
-use super::packet::{read, write, Packet};
+
+use super::packet::read;
 use std::net::TcpListener;
 use std::net::TcpStream;
+use super::packet_router;
 
 pub fn listen() {
     let listener = TcpListener::bind("127.0.0.1:7878").unwrap();
-
+    let mut state = 0;

```

Comment:
This is wrong- the server doesn't have a state, only the connection has a state. For example, a user could be logging in while other players are already playing- then each of those connections would have its own state

Reply:
So we should build a table of connections and then pass the connection itself as a parameter instead of using a mutable state, right ?

Reply:
No- that would technically work but it would break all of our abstractions. Handshake is not a function of all the connections- so passing it all the connections doesn't seem right. I think we need a better solution for state management then what we have now, but the way to fix that short term is to just go back to what we had before with defining the state in the handle_connection method, and we can try to figure out how to cleanly do this later

Reply:
I think for now we should just have each connection defining the state itself, I think as a part of my PR I'll have to find a solution for this (since I need to figure out how to facilitate interconnection communication)

Reply:
Ok so I'll put it back in the handle_connection function